File: service-ai-newtown/db/postgres/models/contractlens_ai_prompt.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from db.postgres.client import Base


class ContractLensAiPrompt(Base):

    __tablename__ = "contractlens_ai_prompt"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        autoincrement=True
    )

    prompt_key: Mapped[str] = mapped_column(
        String(100),
        nullable=False
    )

    prompt_type: Mapped[str] = mapped_column(
        String(50),
        nullable=False
    )

    content: Mapped[str] = mapped_column(
        Text,
        nullable=False
    )

    version: Mapped[int] = mapped_column(
        Integer,
        nullable=False
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False
    )

    created_by: Mapped[str | None] = mapped_column(
        String(100),
        nullable=True
    )

    updated_by: Mapped[str | None] = mapped_column(
        String(100),
        nullable=True
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False
    )

File: service-gateway/src/main/java/com/contractlens/service/db/postgres/dao/ContractLensAiPrompt.java
package com.contractlens.service.db.postgres.dao;

import com.contractlens.common.enums.PromptType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(
        name = "contractlens_ai_prompt",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_ai_prompt_key_version",
                        columnNames = {"prompt_key", "version"}
                )
        }
)
public class ContractLensAiPrompt extends  BaseAuditEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(
            name = "prompt_key",
            nullable = false,
            length = 150
    )
    private String promptKey;

    @Enumerated(EnumType.STRING)
    @Column(
            name = "prompt_type",
            nullable = false,
            length = 50
    )
    private PromptType promptType;

    @Column(
            name = "content",
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String content;

    @Column(
            name = "version",
            nullable = false
    )
    private Integer version;

    @Column(
            name = "is_active",
            nullable = false
    )
    private Boolean isActive = false;

}
